firestore/FirestoreExample/RestaurantDetailViewController.swift

```diff
@@ -142,7 +142,7 @@ class RestaurantDetailViewController: UIViewController, UITableViewDataSource, U
       }
 
       // Error if the restaurant data in Firestore has somehow changed or is malformed.
-      guard let restaurant = restaurantSnapshot.data().flatMap(Restaurant.init(dictionary:)) else {
+      guard let restaurant = try? restaurantSnapshot.data(as: Restaurant.self) else {
```

Comment:
Same as above: since Codable gives us better error messages, we should use them

Reply:
Done.